### PR TITLE
[TF-TRT] Refactoring Top Level __init__ file for TF-TRT

### PR DESCRIPTION
**Pull request by DEKHTIARJonathan**
- Builds on #50782

Applies final cleaning and refactoring on TRT version checking

@bixia1 : for review

### CODE REVIEW

**Comment by DEKHTIARJonathan** [comment]
@bixia1 the refactoring commit is the following: 1cea6a041b19b6265dcde7f4b4650cd75b069386

**Comment by DEKHTIARJonathan** [comment]
@bixia1 the refactoring commit is the following: 3aa8b946c1b85ab2660ce6f47104068150228dcb

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 1]
```diff
@@ -19,5 +19,74 @@
...
 from tensorflow.python.compiler.tensorrt import trt_convert as trt
+from tensorflow.python.compiler.tensorrt import utils as trt_utils
+from tensorflow.python.platform import tf_logging as logging
 # pylint: enable=unused-import,line-too-long
+
+
+if not _pywrap_py_utils.is_tensorrt_enabled():
+  """Check that TensorFlow has been built with TensorRT support enabled.
+  Raises:
+    RuntimeError: if the TensorFlow has not be built with TensorRT support.
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:32]: s/be/been/

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 2]
```diff
@@ -19,5 +19,74 @@
...
+  """
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:33]: Is this just a comment? Why do we even need this comment to repeat what is in the code? Let's delete this few lines.

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 3]
```diff
@@ -19,5 +19,74 @@
...
+      "Tensorflow needs to be built with TensorRT support enabled to allow "
+      "TF-TRT to operate.")
+
+  raise RuntimeError("Tensorflow has not been built with TensorRT support.")
+
+else:
+  """Check compatibility of TensorRT version.
+  Raises:
+    RuntimeError: if the TensorRT library version is incompatible.
+  """
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:44]: Can we just delete these few lines?

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 4]
```diff
@@ -19,5 +19,74 @@
...
+
+
+  def raise_trt_version_deprecated(version_type, trt_version):
+    assert version_type in [
+        "linked", "loaded"
+    ], ("Incorrect value received for version_type: %s. Accepted: ['linked', "
+        "'loaded']") % version_type
+
+    logging.error(
+        "The {version_type} version of TensorRT: `{trt_version}` has now "
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:60]: s/The/The support for/

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 5]
```diff
@@ -19,5 +19,74 @@
...
+    raise_trt_version_deprecated("linked", linked_version)
+
+  if not trt_utils.is_loaded_tensorrt_version_greater_equal(7, 0, 0):
+    raise_trt_version_deprecated("loaded", loaded_version)
+
+  if (loaded_version[0] != linked_version[0] or
+      not trt_utils.is_loaded_tensorrt_version_greater_equal(*linked_version)):
+    logging.error(
+        "Loaded TensorRT %s but linked TensorFlow against TensorRT %s. A few "
+        "requirements must be met:\n"
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:78]: Can we change "A few requirements must be met" to "Not all requirements are met"?

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 6]
```diff
@@ -19,5 +19,74 @@
...
+        "\t-It is required to use the same major version of TensorRT during "
+        "compilation and runtime.\n"
+        "\t-TensorRT does not support forward compatibility. The loaded "
+        "version has to be equal or more recent than the linked version.",
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:82]: Let's s/to be equal or more recent than/to be not later than/

**Comment by gbaned** [comment]
@DEKHTIARJonathan Can you please check @bixia1's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@DEKHTIARJonathan  Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@DEKHTIARJonathan Any update on this PR? Please. Thanks!

**Comment by DEKHTIARJonathan** [comment]
Please do not close

File: tensorflow/python/compiler/tensorrt/__init__.py (hunk 1 above)
Comment (DEKHTIARJonathan) [reply tensorflow/python/compiler/tensorrt/__init__.py:28]: Fixed

File: tensorflow/python/compiler/tensorrt/__init__.py (hunk 4 above)
Comment (DEKHTIARJonathan) [reply tensorflow/python/compiler/tensorrt/__init__.py:56]: Fixed

File: tensorflow/python/compiler/tensorrt/__init__.py (hunk 5 above)
Comment (DEKHTIARJonathan) [reply tensorflow/python/compiler/tensorrt/__init__.py:74]: Fixed

File: tensorflow/python/compiler/tensorrt/__init__.py (hunk 6 above)
Comment (DEKHTIARJonathan) [reply tensorflow/python/compiler/tensorrt/__init__.py:78]: Fixed

**Comment by DEKHTIARJonathan** [comment]
@bixia1 all comments have been addressed. Please review

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 7]
```diff
@@ -15,5 +15,65 @@
...
+
+
+if not _pywrap_py_utils.is_tensorrt_enabled():
+  logging.error(
+      "Tensorflow needs to be built with TensorRT support enabled to allow "
+      "TF-TRT to operate.")
+
+  raise RuntimeError("Tensorflow has not been built with TensorRT support.")
+
+else:
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:32]: There is no need to use "else" if the if-branch will stop with an error. This allows us to reduce the indentation of the code after here.

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 8]
```diff
@@ -15,5 +15,65 @@
...
+  linked_version = _pywrap_py_utils.get_linked_tensorrt_version()
+  loaded_version = _pywrap_py_utils.get_loaded_tensorrt_version()
+
+  logging.info("Linked TensorRT version: %s", str(linked_version))
+  logging.info("Loaded TensorRT version: %s", str(loaded_version))
+
+
+  def raise_trt_version_deprecated(version_type, trt_version):
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:40]: Let's rename the routine to match what it does, prefix it with _ to make it private in the module, unnest it. We mentioned the style guide for nested function in another PR recently.
   _raise_trt_version_no_supported

**Comment by bixia1** [comment]
Please update the PR description, and describe that this PR does.

File: tensorflow/python/compiler/tensorrt/__init__.py (hunk 7 above)
Comment (DEKHTIARJonathan) [reply tensorflow/python/compiler/tensorrt/__init__.py:32]: Fixed

File: tensorflow/python/compiler/tensorrt/__init__.py (hunk 8 above)
Comment (DEKHTIARJonathan) [reply tensorflow/python/compiler/tensorrt/__init__.py:40]: Fixed

**Review by bixia1 — APPROVED** [review]

**Comment by DEKHTIARJonathan** [comment]
@bixia1 I had to add this line: https://github.com/tensorflow/tensorflow/pull/50884/files#diff-0de5f087b61e9a3572b30a6a3ac6f7cbc88f96de81c733334784077646b9f198R51

Otherwise CPU builds would fail. Please re-approve

File: tensorflow/python/compiler/tensorrt/__init__.py [hunk 9]
```diff
@@ -15,5 +15,69 @@
...
+  logging.error(
+      "Tensorflow needs to be built with TensorRT support enabled to allow "
+      "TF-TRT to operate.")
+
+  raise RuntimeError("Tensorflow has not been built with TensorRT support.")
+
+
+# Do not execute the following on a machine that is not linux-based and during
+# bazel build execution.
+if "bazel" not in os.environ["_"] and sys.platform == "linux":
```
Comment (bixia1) [inline tensorflow/python/compiler/tensorrt/__init__.py:51]: You can do "if not build: realy return" to reduce the needed indentation for the code.

**Comment by gbaned** [comment]
@DEKHTIARJonathan Can you please check @bixia1's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@DEKHTIARJonathan Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@DEKHTIARJonathan Any update on this PR? Please. Thanks!

**Pull request closed without merging**